ui/src/validation.ts

```diff
@@ -51,16 +64,34 @@ export const createValidationStore = (constraints: any): ValidationStore => {
     if (errors) {
       update(() => { return { status: ValidationStatus.Error, message: errors.input[0] } })
       return
-    } else {
-      // Check remote
-      await delay(1000).then(() =>
-        update((store) => {
-          // If the input has changed since this request was fired off, don't update
-          if (get(inputStore) !== input) return store
-          return { status: ValidationStatus.Success }
+    }
+
+    // Check remote validation
+    if (remoteValidation) {
+      try {
+        await remoteValidation.promise(input)
```

Comment:
Is this a doubly nested promise? The mix of await and then style promise resolution makes this quite hard to reason about. Is there room to improve and untangle this?

Reply:
Are you thinking along the lines of the event loop architecture? That might look better